File: api/HomeBudget.API/HomeBudget.Application/DTOs/CategoriaDtos/CreateCategoriaDto.cs
using HomeBudget.Domain.Enums;

namespace HomeBudget.Application.DTOs.CategoriaDtos
{
    public class CreateCategoriaDto
    {
        public Guid Id { get; set; }
        public required string Nome { get; set; }
        public required string Descricao { get; set; }

        public required TipoCategoria Finalidade { get; set; }

    }
}

File: web/src/interfaces/CategoriaDtos/CreateCategoriaDto.ts
import { TipoCategoria } from "../../enums/TipoCategoriaEnum";
import { empty_Guid } from "../../utils/guid";

export class CreateCategoriaDto {
  id: string;
  nome: string = "";
  descricao: string = "";
  finalidade: number = TipoCategoria.Despesa;

  constructor(
    id: string = empty_Guid,
    nome: string = "",
    descricao: string = "",
    finalidade: number = TipoCategoria.Despesa,
  ) {
    this.id = id;
    this.nome = nome;
    this.descricao = descricao;
    this.finalidade = finalidade;
  }
}
